api-resources.md

```diff
@@ -0,0 +1,337 @@
+
+##### Alternatives
+
+An HPKE config has an ID, so we could use a path `hpke_configs/hpke-config-id`. However, a client should never need to get any HPKE config except for the "current" one, so does it make sense to expand the API surface with a path param for HPKE config ID?
+
+`GET /hpke_config` could redirect to `/hpke_config/current-hpke-config-id`, which would make this mostly transparent to clients, assuming they handle 3xx redirects properly.
+
+Instead of providing the task ID as a query parameter, we could also include it in the path, like `/tasks/{task-id}/hpke_config`, on the premise that the task ID is part of the resource's permanent identity. However, it's awkward to solve the global HPKE config case with this formulation. We could include a special task ID "global" so that clients could query `/tasks/global/hpke_config`.
+
+#### Representation
+
+Uses the existing [`struct HpkeConfig`](https://www.ietf.org/archive/id/draft-ietf-ppm-dap-02.html#section-4.3.1-6).
+
+```
+struct {
+  HpkeConfigId id;
+  HpkeKemId kem_id;
+  HpkeKdfId kdf_id;
+  HpkeAeadKdfId aead_id;
+  HpkePublicKey public_key;
+} HpkeConfig;
+```
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+The response body is a `struct HpkeConfig` representing the `hpke_config`.
+
+### Reports
+
+#### Path
+
+`/tasks/{task-id}/reports/{report-id}`
+
+`report-id` uniquely identifies a report in a task. It is chosen by the client when uploading a report.
+
+#### Representation
+
+```
+struct {
+  Time time;
+  Extension extensions<0..2^16-1>;
+} ReportMetadata;
+
+struct {
+  ReportMetadata metadata;
+  opaque public_share<0..2^32-1>;
+  HpkeCiphertext encrypted_input_shares<1..2^32-1>;
+} Report;
+```
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+The response body is a `struct Report` representing the report.
+
+##### PUT
+
+Idempotent upload of a report to an aggregator. The request body is a `struct Report`.
+
+This is analogous to DAP-02's `POST /upload`.
+
+### An aggregation job
+
+Only the helper supports this resource.
+
+#### Path
+
+`/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}`
+
+`aggregation-job-id` uniquely identifies an aggregation job. It is chosen by the leader when it dispatches a job to the helper.
+
+#### Representation
+
+```
+enum {
+  continued(0),
+  finished(1),
+  failed(2),
+  (255),
+} PrepareStepState;
+
+struct {
+  ReportID report_id;
+  PrepareStepState prepare_step_state;
+  select (PrepareStep.prepare_step_state) {
+    case continued: opaque prep_msg<0..2^32-1>; /* VDAF preparation message */
+    case finished: Empty;
+    case failed: ReportShareError;
+  }
+} PrepareStep;
+
+struct {
+  u16 round;
+  PrepareStep prepare_steps<1..2^32-1>;
+} AggregateJob
+```
+
+See "POST", below, for explanation and justification of the `round` field.
+
+#### Required HTTP methods
+
+##### GET/HEAD
+
+The response body is a `struct AggregateJob` representing the current state of the reports in the job.
```

Comment:
Who needs to implement this: the Leader, Helper, or both? Presumably just the Helper? How long do we need to keep this around?

Reply:
Just the helper, as noted above, but we do need to--

- [ ] add some language discussing how long servers should have to respond to GET aggregate jobs, aggregate shares, collections, considering task expiration and DELETE requests against these